marklogic-client-api/src/main/java/com/marklogic/client/datamovement/impl/BatcherImpl.java

```diff
@@ -24,7 +24,7 @@
 public abstract class BatcherImpl implements Batcher {
   private String jobName = "unnamed";
   private String jobId = null;
-  private int batchSize = 100;
+  private int batchSize = 0;
```

Comment:
I don't think this change can be made at this level; a QueryBatcherTest test is failing due to a divide-by-zero change. At minimum, I think this change needs to happen within WriteBatcherImpl so that it does not impact RowBatcher or QueryBatcher. 

Also, there's logging indicating that this is immediately set to a batch size of 1- "11:00:52.984 [main] WARN  c.m.c.d.impl.WriteBatcherImpl - batchSize should be 1 or greater--setting batchSize to 1". We wouldn't want that appearing any time the user doesn't explicitly set batch size. 

I'll ping you internally about this, as I'm not sure that the current behavior is a problem - i.e. the test you wrote can easily be made to pass by using `flushAndWait` instead of `awaitCompletion`.